synapse/storage/databases/main/relations.py

```diff
@@ -682,10 +682,20 @@ async def get_bundled_aggregations(
             A map of event ID to the bundled aggregation for the event. Not all
             events may have bundled aggregations in the results.
         """
+        # The already processed event IDs. Tracked separately from the result
+        # since the result omits events which do not have bundled aggregations.
+        seen_events = set()
```

Comment:
I was very much hoping to be able to do `events = set(events)`, but I don't think that would work as events aren't comparable as far as I can tell.